src/use_event_listener/index.jsx

```diff
@@ -1,18 +1,26 @@
-import { useEffect } from 'react';
+import { useEffect, useRef } from 'react';
 
 export const useEventListener = (
   ref,
   type,
-  listener,
-  options = { passive: true },
+  handler,
+  { passive, capture, once } = { passive: true },
 ) => {
+  const node = ref.current;
+  const handlerRef = useRef(null);
+
+  handlerRef.current = handler;
+
   useEffect(() => {
-    if (!ref.current) return;
+    if (!node) return;
+
+    const listener = (...args) => handlerRef.current(...args);
+    const options = { passive, capture, once };
 
-    ref.current.addEventListener(type, listener, options);
+    node.addEventListener(type, listener, options);
 
     return () => {
-      ref.current.removeEventListener(type, listener);
+      node.removeEventListener(type, listener, options);
     };
-  }, [type, listener, options]);
+  }, [type, node, passive, capture, once]);
```

Comment:
this will be executed on every rerender, probably not what you intended

Reply:
Why though?

Reply:
AFAIK ref.current will change on re-renders (flash null/node)

Reply:
but if it is a different element after re-render, it needs to have the listener attached?

Reply:
ah good point